### PR TITLE
Enhance Tensor print

### PR DESCRIPTION
**Pull request by zhoonit**
**Changes proposed in this PR:**
- Add tensor address to the print
- Handle the case when tensor is too large
- Add test

See also: #270

**Self evaluation:**
1. Build test: [X]Passed [ ]Failed [ ]Skipped
2. Run test: [X]Passed [ ]Failed [ ]Skipped

Signed-off-by: Jihoon Lee <jhoon.it.lee@samsung.com>

### CODE REVIEW

File: nntrainer/src/tensor.cpp [hunk 1]
```diff
@@ -783,17 +783,24 @@ Tensor Tensor::apply(std::function<float(float)> f) const {
...
+  unsigned int len = length();
+
+  if (len > 100) {
+    out << '[' << data[0] << ' ' << data[1] << ' ' << data[2] << " ... "
+        << data[len - 3] << ' ' << data[len - 2] << ' ' << data[len - 1] << ']'
+        << std::endl;
+    return;
+  }
+
+  unsigned int i, j, k, l;
```
Comment (wooksong) [inline nntrainer/src/tensor.cpp:798]: 1. If these iterator variables are used only for the following 'for' loops, you can move them into corresponding 'for' statements.

2. Comments for the following 'for' statements: 
dim.batch(), dim.channel(), dim.height(), dim.width(), and dim.getFeatureLen() would be invoked multiple times for no reasons. Please use variables for them outside the following 'for' loops.

Comment (zhoonit) [reply nntrainer/src/tensor.cpp:798]: Q. Regrading 2. those are inline functions. Does using a variable outside of the loop make a difference?

Comment (zhoonit) [reply nntrainer/src/tensor.cpp]: Updated for 1. Thanks :)

File: nntrainer/src/tensor.cpp [hunk 2]
```diff
@@ -783,17 +783,25 @@ Tensor Tensor::apply(std::function<float(float)> f) const {
...
-          out << data[k * dim.getFeatureLen() + l * dim.width() * dim.height() +
-                      i * dim.width() + j]
-              << " ";
+
+  unsigned int len = length();
+
+  out << dim;
+
+  if (len > 100) {
+    out << '[' << data[0] << ' ' << data[1] << ' ' << data[2] << " ... "
```
Comment (kparichay) [inline nntrainer/src/tensor.cpp:794]: How about skip printing data in the model summary. If someone needs data, they can directly inquire tensor.

**Comment by kparichay** [comment]
Other than @wooksong comments, LGTM.

File: nntrainer/src/tensor.cpp (hunk 1 above)
Comment (wooksong) [reply nntrainer/src/tensor.cpp]: A for 2. https://isocpp.org/wiki/faq/inline-functions#inline-and-perf

Comment (zhoonit) [reply nntrainer/src/tensor.cpp]: Thanks you for the explanation :)

**Pull request merged**